frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -278,17 +283,17 @@ pub mod pallet {
             }
 
             let force_new_era = Self::force_era().eq(&Forcing::ForceNew);
-            let blocks_per_era = T::BlockPerEra::get();
             let previous_era = Self::current_era();
+            let next_era_starting_block = Self::next_era_starting_block();
 
             // Value is compared to 1 since genesis block is ignored
-            if now % blocks_per_era == BlockNumberFor::<T>::from(1u32)
-                || force_new_era
-                || previous_era.is_zero()
-            {
+            if now == next_era_starting_block || force_new_era || previous_era.is_zero() {
```

Comment:
Small change of requirements - if we `pause` the pallet, it'd be good if normal operation was resumed as fast as possible once it's been unpaused. There's no need to wait for double or triple amount of blocks per era to pass before triggering a new one.

We can achieve this by changing the condition to `>=`.